source/write/bulk-write.txt

```diff
@@ -0,0 +1,346 @@
+- Replacement document that specifies the new fields and values to insert
+
+The following example creates an instance of ``ReplaceOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To replace multiple documents, you must create an instance of
+``ReplaceOneModel`` for each document.
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To delete a document, create an instance of ``DeleteOneModel`` and pass a
+query filter specifying the document you want to delete. A
+``DeleteOneModel`` instance provides instructions to delete
+only *the first* document that matches your query filter.
+
+The following example creates an instance of ``DeleteOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To delete multiple documents, create an instance of ``DeleteManyModel`` and pass a
+query filter specifying the document you want to delete. An instance of
+``DeleteManyModel`` provides instructions to remove *all* documents that
+match your query filter.
+
+The following example creates an instance of ``DeleteManyModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: scala
+   :copyable:
+   :dedent:
+
+.. _scala-bulkwrite-method:
+
+Perform the Bulk Operation
+--------------------------
+
+After you define a model instance for each operation you want to perform,
+pass a list of these instances to the ``bulkWrite()`` method.
+By default, the method runs the operations in the order
+specified by the list of models.
+
+The following example performs multiple write operations by using the
+``bulkWrite()`` method:
+
+.. io-code-block::
+
+   .. input:: /includes/write/bulk-write.scala
+      :start-after: start-bulk-write-mixed
+      :end-before: end-bulk-write-mixed
+      :language: scala
+      :dedent:
+
+   .. output::
+      
+      Success: AcknowledgedBulkWriteResult{insertedCount=1, matchedCount=1, removedCount=1,
+      modifiedCount=1, upserts=[], inserts=[BulkWriteInsert{index=0, id=BsonObjectId{value=...}}]}
+      Completed
+
+If any of the write operations fail, the {+driver-short+} raises a
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` item that includes the
+operation that failed and details about the exception.
+
+.. note::
+
+   When the driver runs a bulk operation, it uses the write concern of the
+   target collection. The driver reports all write concern errors after
+   attempting all operations, regardless of execution order.
+
+Customize Bulk Write Operation
+------------------------------
+
+The ``bulkWrite()`` method optionally accepts a parameter which
```

Comment:
S: you can introduce BulkWriteOptions here, since you mention it in the next paragraph
```suggestion
The ``bulkWrite()`` method optionally accepts a ``BulkWriteOptions`` parameter, which
```